### DISCUSSION TITLE
Add support for nested daily-note locations (e.g. `2020/10/04.md` or `2020-10/04.md`)

### DISCUSSION THREAD

**Discussion by jplattel** [liamcain/obsidian-calendar-plugin#21] [Ideas]
Hi Liam!

Thanks for this wonderful plugin! I have a feature request that doesn't have any priority so feel free to judge if you want to solve this. Here's the thing:

I'm grouping every daily note in a folder after a month is finished:

<img width="201" alt="Screenshot 2020-11-02 at 22 39 43" src="https://user-images.githubusercontent.com/517775/97922179-58444d00-1d5c-11eb-9db8-2be356ec8029.png">

The daily notes of the current month are in the root of the `daily` folder. 

After moving the notes the plugin can't seem to find the notes anymore, thus asking me to create a new one. Is this something that could be solved? It might be a hard problem to figure out the linking to each file...

**Comment by liamcain** [comment]
Hi, thanks for the feature request! This is certainly doable, but my main focus right now is to maintain compatibility with the Daily Notes plugin.

I'll likely revisit that down the road to enable more custom workflows like yours, so I'll definitely keep this in mind.

**Comment by liamcain** [comment]
# 👋 What's your workflow

Converted this to a **Discussion** (mostly to try out the feature, but also to get more input). I'm curious about people's workflow for this. There are basically 2 aspects of the plugin that need to adapt to support this behavior:
1.) Finding existing notes
2.) Creating new notes

Both of these have caveats so here are my open questions.

## Open Design Questions!
### How flexible?

Should the plugin just look inside all folders within the daily-note folder until it finds a daily note? (maximum flexibility)
<img width="193" alt="image" src="https://user-images.githubusercontent.com/693981/101656968-592e6580-3a11-11eb-9021-2f11be574f68.png">

Or should it have a defined format path (e.g. `YYYY/MM/YYYY-MM-DD`)?
<img width="174" alt="image" src="https://user-images.githubusercontent.com/693981/101657663-23d64780-3a12-11eb-829d-e1816813d3eb.png">


###  Where do the files get created?
When creating files, would you rather have the files in their "final folder" or is moving them into their monthly folder an important part of your workflow?

### Custom Settings?
A big challenge is that I'm building around the existing daily-notes plugin. I'm not sure I could reuse the existing daily-notes setting to support this kind of change. Would it be super confusing to provide another set of settings in the Calendar plugin itself for more advanced configuration (folder/path/template)?

**For Reference: Existing Daily Note Settings**
<img width="940" alt="image" src="https://user-images.githubusercontent.com/693981/101660434-2b4b2000-3a15-11eb-829b-f88c90d40e38.png">

**Is this flexible enough?**
<img width="945" alt="image" src="https://user-images.githubusercontent.com/693981/101660744-80873180-3a15-11eb-9884-b38fc34ee399.png">
- How do I denote that YYYY-MM in the folder path is a date format?
- This will break the existing autocomplete which would be a confusing UX
- Since it's reusing the Daily Notes plugin, I can't add any additional help text or clarity to it so users will likely get lost

**Forking the experience**
<img width="914" alt="image" src="https://user-images.githubusercontent.com/693981/101661548-6732b500-3a16-11eb-9152-78b6736098c2.png">
- Other plugins rely on the daily note settings too. If you are using custom settings that live in Calendar, then other plugins will be out of sync.

### Okay to break compatibility?
In your existing workflow, do you use the existing daily-notes commands at all (go to next day/ go to previous day)?

---

The simplest option would be to keep the existing functionality in place (creating it at the root of the daily-note folder) then just looking inside all the subfolders when finding daily notes. So the question becomes, is this enough for supporting this type of workflow?

**Reply by liamcain** [reply]
One other idea I had: If I just added support for the nested subfolders, another plugin could basically work like [Hazel](https://www.noodlesoft.com/) and make sure the files are in the right spot. I think that could be the easiest.

**Reply by jplattel** [reply]
Hey @liamcain making it a discussion is the way to go! 

Personally I only need the first part, finding the right notes. I'm moving them manually and don't mind doing that once a month. If push comes to shove an plugin like Hazel might the way to go, but that is out of scope for this extension.

As for the daily-note commands I'm using only the 'open today's note' command, as I'm using next/prev day in the note itself. 

The command palette has fuzzy searching and that supports these nested folders, so maybe it's possible to re-use that in a way?

**Reply by liamcain** [reply]
Totally! I think it's important to draw a line in the sand for what the plugin is responsible for. 

Of all the similar workflows that I've heard, I think your use case is the easiest to implement 😄. I'm hoping to get a sense for how this change works in the overall plugin ecosystem to future-proof this a bit.

The gap (creating daily-notes in nested folders) might be fillable via another plugin, in which case... 🎉. We can call it a day.

---

I appreciate your thoughts on this, feels a lot more feasible now!

**Reply by Aegiswiz** [reply]
Thanks Liam. Anything in "Is this flexible enough" works for everything I need and more.

I don't routinely create notes w/the calendar. Primarily just navigation thought the daily notes. After reviewing how my work flow went in last two programs back to 2015, and Obsidian through 3 changes in making it suit my needs, (too much trouble & too much data to try a 4th time) Just navigation... Maybe to see if I wrote a lot on any give day, great idea with that word count thing BTW. 

The weekly thing may turn out to be very helpful as well if you decide to make the change to above. I thought the weekly option might contain a fix for me but in weeks like December, week 49 has Nov 29 & 30 which doesn't show up, so that ends up not working as is. Then I'd have a complicated "Template" issue I've not resolved yet, so...

I do have a question if you decide it's worth changing. In your template suggestion you have ![[{{sunday: YYYY-MM-DD}}]] for example. (Thanks for that help!) And that can work really great, although I'm not sure what the day in there does, it does nothing when creating the weekly note that I can discern, but it did get me thinking about a template that would be wonderful for me. I tried this... [[{{sunday: YYYY-MM-DD#}}]] with and without the day, with and without the colon.  What I was trying to accomplish is # brings up the ONLY heading in my daily notes (95% of the time) and that heading has all the info I need.  I don't have a clue how to make that work in a template. BUT we ALSO have our new "alias" trick.  I can make the only heading into an alias.  Can templates handle inserting the/an alias?  If it can, I just need remove the ! embedding, and I have exactly what I need in a weekly note.

And thanks again Liam.  I knew you weren't interested in messing with this & I know you've now put a lot more time and thought into it.  I had just accepted you weren't likely to change your mind.  So regardless of what you end up doing (I quit programming decades ago) I do appreciate the work you've done just on 2 folks hoping it might change.  I do dearly love all the folks working on Obsidian & do my best to encourage that effort continues.

**Comment by liamcain** [comment]
cc: @ryanjamurphy @tgrosinger Curious if either of you have thoughts on this!

**Reply by tgrosinger** [reply]
Yeah, I've been thinking about it a little bit. I think the biggest benefit for the smallest change would be to look for daily notes in all subfolders of the configured daily note directory. Then I am free to move them around (for example, at the end of each month) without breaking anything.

Adding an option to specify a nested folder path in the daily note folder would be the next step, as it would make it so I don't need to move the notes later. If this option is implemented though, I would still look for notes in all subdirectories.

**Comment by liamcain** [comment]
Thanks for all the input! In version 1.4.5, the calendar will now find all daily notes nested inside the daily note folder. Notes will still be created at the root of the daily note folder but if you move it into a subfolder, it won't get lost. 

Let me know if you have any issues with it!

**Reply by Aegiswiz** [reply]
Just saw your message.  Since I change the daily notes to the subfolder as the daily folder at the end of each month the dailies always appear for the current month in the calendar. i.e. They are created in zz-Dailies/20-12. Last month zz-Dailies/20-11, October, I used a different naming convention so most those don't work. Hoping I understood your message. I made zz-Dailies/ my daily notes (default).  I then moved all my daily docs into that folder, one month at a time.  Restarted Obsidian.  Of course the ones I moved to the default folder showed up.  When I moved them back to their subfolder for their month, they vanished.  I had hoped that having been in the new default folder they would be picked up in the subfolder I returned them to. That was not what I experienced. They only show up in the designated default folder. Honestly, it is most likely I'm doing something wrong. Thanks for working on this @liamcain.  Your attention to the wishes of your calendar users is extraordinary.

**Reply by liamcain** [reply]
Hmm, I think I botched something with the release. I can repro the behavior you are seeing. Will let you know when I have a fix.

**Reply by liamcain** [reply]
Okay, sorry for the hiccup. Looks like I my release didn't include the newest code. So I just pushed v1.4.6 which should _actually_ support the nested folders.

As for the workflow, you should be able to keep the daily notes folder as `zz-Dailies` and then you'll just need to move newly created files into the latest subfolder (e.g. `20-12`) manually.

**Comment by jplattel** [comment]
Just upgraded, will let you know if I find any issues. Thanks for taking the time to build this @liamcain!

**Comment by arnoesterhuizen** [comment]
With regards to your interface:

You could have date format contain the date variables with any combination of dashes and slashes (e.g. YYYY-MM/YYYY-MM-DD), and have "new file location" still point to an existing folder, as is.

When you create a new daily note, it would simply append "new file location" to "date format" and ".md", create the path if it does not exist without messing with the format or breaking the UI or expectations.
